src/GafferSceneUI/RotateTool.cpp

```diff
@@ -172,12 +230,62 @@ bool RotateTool::dragMove( const GafferUI::Gadget *gadget, const GafferUI::DragD
 	return true;
 }
 
-bool RotateTool::dragEnd()
+bool RotateTool::handleDragEnd()
 {
 	TransformTool::dragEnd();
 	return false;
 }
 
+bool RotateTool::buttonPress( const GafferUI::ButtonEvent &event )
+{
+	if( event.buttons != ButtonEvent::Left
+		|| !activePlug()->getValue()
+		|| !getTargetedMode()
+	) {
+		return false;
+	}
+
+	// In targeted mode, we orient the selection so -z aims towards the clicked point.
+	//
+	// We always return true to prevent the SelectTool defaults.
+
+	GafferScene::ScenePlug::ScenePath _;
+	Imath::V3f targetPos;
+
+	const SceneView *sv = static_cast<const SceneView *>( view() );
+	const Gadget *g = sv->viewportGadget()->getPrimaryChild();
+	const SceneGadget* sg = static_cast<const SceneGadget *>( g );
+	if( !sg->objectAt( event.line, _, targetPos ) )
+	{
+		return true;
+	}
+
+	UndoScope undoScope( selection()[0].transformPlug->ancestor<ScriptNode>(), UndoScope::Enabled );
+
+	for( const auto &s : selection() )
+	{
+		Context::Scope scopedContext( s.context.get() );
+
+		V3f currentYAxis, currentZAxis;
+		const M44f worldTransform = s.upstreamScene->fullTransform( s.upstreamPath );
```

Comment:
I haven't tested, but I suspect this has the same bug as the one I noted in TranslateTool.

Reply:
Should be fixed in https://github.com/GafferHQ/gaffer/pull/3403/commits/5bcf51b26c2cbc7d19e045f0a34c38ccab013ca6